iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/json/NumericLiteralImplTest.java

```diff
@@ -23,25 +23,24 @@
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
 import org.sonar.iac.arm.parser.ArmParser;
-import org.sonar.iac.arm.tree.api.File;
 import org.sonar.iac.arm.tree.api.Property;
-import org.sonar.iac.arm.tree.api.ResourceDeclaration;
 import org.sonar.iac.common.extension.ParseException;
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.sonar.iac.arm.ArmAssertions.assertThat;
-import static org.sonar.iac.arm.tree.impl.json.PropertyTestUtils.getCode;
+import static org.sonar.iac.arm.tree.impl.json.PropertyTestUtils.LINE_OFFSET;
+import static org.sonar.iac.arm.tree.impl.json.PropertyTestUtils.parseProperty;
 
 class NumericLiteralImplTest {
   private final ArmParser parser = new ArmParser();
 
   @Test
   void shouldParseNumericValue() {
-    String code = getCode("\"numeric\": 7");
-    File tree = (File) parser.parse(code, null);
-
-    Property numericProperty = ((ResourceDeclaration) tree.statements().get(0)).properties().get(0);
-    assertThat(numericProperty.value()).asNumericLiteral().hasValue(7);
+    Property numericProperty = parseProperty(parser, "\"numeric\": 7");
+    assertThat(numericProperty.value())
+      .asNumericLiteral()
+      .hasValue(7)
+      .hasRange(LINE_OFFSET + 1, 11, LINE_OFFSET + 1, 12);
   }
```

Comment:
IMO more detailed tests for the following methods are needed here: `getKind()`, `children()`, and `parent()`. Similar in other test classes in this PR.